msgraph/core/graph_client_factory.py

```diff
@@ -0,0 +1,117 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import List, Optional
+
+import httpx
+from kiota_abstractions.authentication import AccessTokenProvider
+from kiota_http.kiota_client_factory import KiotaClientFactory
+from kiota_http.middleware import AsyncKiotaTransport
+from kiota_http.middleware.middleware import BaseMiddleware
+
+from ._constants import DEFAULT_CONNECTION_TIMEOUT, DEFAULT_REQUEST_TIMEOUT
+from ._enums import APIVersion, NationalClouds
+from .middleware import (
+    GraphAuthorizationHandler,
+    GraphMiddlewarePipeline,
+    GraphRedirectHandler,
+    GraphRetryHandler,
+    GraphTelemetryHandler,
+)
+
+
+class GraphClientFactory(KiotaClientFactory):
+    """Constructs httpx.AsyncClient instances configured with either custom or default
+    pipeline of graph specific middleware.
+    """
+
+    def __init__(
+        self,
+        api_version: APIVersion,
+        base_url: NationalClouds,
+        timeout: httpx.Timeout,
+        client: Optional[httpx.AsyncClient],
+    ):
+        """Class constructor accepts a user provided client object and kwargs to configure the
+        request handling behaviour of the client
+
+        Args:
+            api_version (APIVersion): The Microsoft Graph API version to be used, for example
+                                    `APIVersion.v1` (default). This value is used in setting
+                                    the base url for all requests for that session.
+            base_url (NationalClouds): a supported Microsoft Graph cloud endpoint.
+            timeout (httpx.Timeout):Default connection and read timeout values for all session
+                                    requests.Specify a tuple in the form of httpx.Timeout(
+                                    REQUEST_TIMEOUT, connect=CONNECTION_TIMEOUT),
+            client (Optional[httpx.AsyncClient]): A custom AsynClient instance from the
+                                    python httpx library
+        """
+        self.api_version = api_version
+        self.base_url = base_url
+        self.timeout = timeout
+        self.client = client
+
+    def create_with_default_middleware(
```

Comment:
My python is not very good, but I just wanted to make sure those methods are static?

Reply:
They now are, as per https://github.com/microsoftgraph/msgraph-sdk-python-core/pull/141/commits/d690941044fbc46530ba9226bd8f110050c5c179